### DISCUSSION TITLE
Slint enums lose underscores in Rust

### DISCUSSION THREAD

**Discussion by otiv-wannes-vanleemput** [slint-ui/slint#5170] [Q&A] [answered]
Im using a Slint enum to control the state the UI is in. It looks like this:
```
export enum FooState {
    HELLO_STATE,
    HAPPY_STATE,
    SAD_STATE,
}
```

In Rust, the FooState enum loses the underscores, for example:

```
let foo_state = FooState.HAPPYSTATE
```

Am I missing the proper casing for enums in Slint? I'm planning to add quite a lot of states, losing the underscores hurts readability, and I don't like that my Rust names don't match the Slint names.

**Comment by ogoffart** [comment, marked as answer]
Enums are converted to camel case. So `hello_state` becomes `HelloState` which match the convention in rust.
In Slint the convention for enums is to use lower case snake/kebab case. So The idomatic way tyo do it would be

```slint
export enum FooState {
    hello-state,
    happy-state,
    sad-state,
}
```

Which would be in rust:

```rust
let foo_state = FooState::HappyState;
```

Admittedly, we should probably have kept the underscore when the name was in upper case. But now that would be a breaking change unfortunately.  (Unless we add compiler option or attribute for this kind of configuration)

**Reply by otiv-wannes-vanleemput** [reply]
Thanks! Changing it to keep underscores probably isn't high priority for you. I'm happy to use lower snake case + camel case.
